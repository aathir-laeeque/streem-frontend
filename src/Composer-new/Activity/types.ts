import {
  Activity,
  ActivityType,
  Checklist,
  Stage,
  Task,
} from '../checklist.types';

export type ActivityProps = {
  activity: Activity;
  taskId: Task['id'];
};

export type AddNewActivityArgs = {
  activityType: ActivityType;
  checklistId: Checklist['id'];
  orderTree: Activity['orderTree'];
  stageId: Stage['id'];
  taskId: Task['id'];
};

export type DeleteActivityArgs = {
  activityId: Activity['id'];
  taskId: Task['id'];
  stageId: Stage['id'];
};

export enum ActivityErrors {
  E405 = 'PROVIDE_REASON_FOR_YES_NO_ACTIVITY',
  E406 = 'PROVIDE_REASON_FOR_PARAMETER_ACTIVITY_OFF_LIMITS',
  E407 = 'YES_NO_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E408 = 'YES_NO_ACTIVITY_SHOULD_HAVE_EXACTLY_TWO_OPTIONS',
  E409 = 'YES_NO_ACTIVITY_TITLE_CANNOT_BE_EMPTY',
  E410 = 'MULTISELECT_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E411 = 'MULTISELECT_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E412 = 'SINGLE_SELECT_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E413 = 'SINGLE_SELECT_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E414 = 'CHECKLIST_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E415 = 'CHECKLIST_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E416 = 'PARAMETER_ACTIVITY_VALUE_CANNOT_BE_EMPTY',
  E417 = 'PARAMETER_ACTIVITY_VALUE_INVALID',
  E418 = 'PARAMETER_ACTIVITY_OPERATOR_CANNOT_BE_EMPTY',
  E419 = 'MATERIAL_ACTIVITY_NAME_CANNOT_BE_EMPTY',
  E420 = 'MATERIAL_ACTIVITY_LIST_CANNOT_BE_EMPTY',
  E421 = 'MATERIAL_ACTIVITY_CANNOT_BE_MANDATORY',
  E422 = 'INSTRUCTION_ACTIVITY_TEXT_CANNOT_BE_EMPTY',
  E423 = 'INSTRUCTION_ACTIVITY_CANNOT_BE_MANDATORY',
}
