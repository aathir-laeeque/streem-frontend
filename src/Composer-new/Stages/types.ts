import { Stage } from '../checklist.types';

export type StageCardProps = {
  index: number;
  isActive: boolean;
  isFirstItem: boolean;
  isLastItem: boolean;
  stage: Stage;
};

export type ReOrderArgs = {
  from: number;
  id: Stage['id'];
  to: number;
};
