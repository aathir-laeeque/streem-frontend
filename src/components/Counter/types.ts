import { decrement, increment, reset } from './actions';

export interface CounterState {
  readonly count: number;
}

export enum CounterActionTypes {
  INCREMENT = '@@test/INCREMENT',
  DECREMENT = '@@test/DECREMENT',
  RESET = '@@test/RESET',
}

export type CounterAction = ReturnType<
  typeof increment | typeof decrement | typeof reset
>;
