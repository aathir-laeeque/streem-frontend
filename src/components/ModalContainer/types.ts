import {
  openModalAction,
  closeModalAction,
  closeAllModalAction,
} from './actions';

export enum ModalNames {
  CREATE_JOB_MODAL = 'CREATE_JOB_MODAL',
  JOB_USER_ASSIGN = 'JOB_USER_ASSIGN',
}

export interface Modal {
  type: string;
  props: Record<string, any>;
}

export interface ModalContainerProps {
  currentModals: Modal[];
  closeModal: () => void;
  closeAllModals: () => void;
}

export enum ModalContainerAction {
  OPEN_MODAL = '@@modal/Container/OPEN_MODAL',
  CLOSE_MODAL = '@@modal/Container/CLOSE_MODAL',
  CLOSE_ALL_MODAL = '@@modal/Container/CLOSE_ALL_MODAL',
}

export interface ModalContainerState {
  readonly currentModals: { type: string; props: Record<string, any> }[];
}

export type ModalContainerActionType = ReturnType<
  typeof openModalAction | typeof closeModalAction | typeof closeAllModalAction
>;
