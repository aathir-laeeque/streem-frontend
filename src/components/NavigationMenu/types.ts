export interface MenuItem {
  icon: React.ReactType;
  name: string;
  path: string;
}
