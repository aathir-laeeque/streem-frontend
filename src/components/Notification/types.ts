import { showNotification } from './actions';

export enum NotificationAction {
  SHOW_NOTIFICATION = '@@notification/SHOW_NOTIFICATION',
}

export type NotificationActionType = ReturnType<typeof showNotification>;

export enum NotificationType {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning',
}
