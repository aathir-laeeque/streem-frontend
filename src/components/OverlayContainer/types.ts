import {
  openOverlayAction,
  closeOverlayAction,
  closeAllOverlayAction,
  updatePropsAction,
} from './actions';

export enum OverlayNames {
  ADD_STOP = 'ADD_STOP',
  ASSIGNED_USER_DETAIL = 'ASSIGNED_USER_DETAIL',
  ASSIGNMENT_SUCCESS = 'ASSIGNMENT_SUCCESS',
  COMPLETE_JOB_WITH_EXCEPTION = 'COMPLETE_JOB_WITH_EXCEPTION',
  COMPLETE_TASK_WITH_EXCEPTION = 'COMPLETE_TASK_WITH_EXCEPTION',
  CONFIRMATION_MODAL = 'CONFIRMATION_MODAL',
  CREATE_JOB_MODAL = 'CREATE_JOB_MODAL',
  JOB_USER_ASSIGN = 'JOB_USER_ASSIGN',
  MEDIA_DETAIL = 'MEDIA_DETAIL',
  SIGNATURE_MODAL = 'SIGNATURE_MODAL',
  SIMPLE_CONFIRMATION_MODAL = 'SIMPLE_CONFIRMATION_MODAL',
  START_JOB_MODAL = 'START_JOB_MODAL',
  SKIP_TASK_MODAL = 'SKIP_TASK_MODAL',
  TASK_ERROR_CORRECTION = 'TASK_ERROR_CORRECTION',
  TASK_MEDIA = 'TASK_MEDIA',
  TASK_USER_ASSIGNMENT = 'TASK_USER_ASSIGNMENT',
  CHECKLIST_REVIEWER_ASSIGNMENT = 'CHECKLIST_REVIEWER_ASSIGNMENT',
  CHECKLIST_REVIEWER_ASSIGNMENT_SUCCESS = 'CHECKLIST_REVIEWER_ASSIGNMENT_SUCCESS',
  SUBMIT_REVIEW_MODAL = 'SUBMIT_REVIEW_MODAL',
  CHECKLIST_REVIEWER_SUBMIT_SUCCESS = 'CHECKLIST_REVIEWER_SUBMIT_SUCCESS',
  CHECKLIST_SENT_TO_AUTHOR_SUCCESS = 'CHECKLIST_SENT_TO_AUTHOR_SUCCESS',
  AUTHORS_DETAIL = 'AUTHORS_DETAIL',
  REVIEWERS_DETAIL = 'REVIEWERS_DETAIL',
  CHECKLIST_REVIEWER_ASSIGNMENT_POPOVER = 'CHECKLIST_REVIEWER_ASSIGNMENT_POPOVER',
  TASK_USERS_ASSIGNMENT = 'TASK_USERS_ASSIGNMENT',
  TIMED_TASK_CONFIG = 'TIMED_TASK_CONFIG',
  INITIATE_SIGNOFF = 'INITIATE_SIGNOFF',
  SIGN_OFF_PROGRESS = 'SIGN_OFF_PROGRESS',
  PASSWORD_INPUT = 'PASSWORD_INPUT',
  SESSION_EXPIRE = 'SESSION_EXPIRE',
  SIGN_OFF_STATE = 'SIGN_OFF_STATE',
  SIGN_COMPLETED_TASKS = 'SIGN_COMPLETED_TASKS',
  SIGNNING_NOT_COMPLETE = 'SIGNNING_NOT_COMPLETE',
  PARAMETER_APPROVAL = 'PARAMETER_APPROVAL',
  EDITING_DISABLED = 'EDITING_DISABLED',
  CHECKLIST_INFO = 'CHECKLIST_INFO',
  SIGN_OFF_INITIATED_SUCCESS = 'SIGN_OFF_INITIATED_SUCCESS',
  SIGN_OFF_SUCCESS = 'SIGN_OFF_SUCCESS',
  RELEASE_SUCCESS = 'RELEASE_SUCCESS',
  SHOW_COMPLETED_JOB_WITH_EXCEPTION_INFO = 'SHOW_COMPLETED_JOB_WITH_EXCEPTION_INFO',
  REVISION_ERROR = 'REVISION_ERROR',
}

export interface CommonOverlayProps<T> {
  type: OverlayNames;
  props?: T;
  key: string;
  closeOverlay: () => void;
  closeAllOverlays: () => void;
  popOverAnchorEl?: Element | ((element: Element) => Element);
}

export interface Overlay {
  type: OverlayNames;
  props?: Record<string, any>;
  popOverAnchorEl?: Element | ((element: Element) => Element);
}

export interface OverlayContainerProps {
  currentOverlays: Overlay[];
  closeOverlay: () => void;
  closeAllOverlays: () => void;
}

export enum OverlayContainerAction {
  OPEN_OVERLAY = '@@overlay/Container/OPEN_OVERLAY',
  CLOSE_OVERLAY = '@@overlay/Container/CLOSE_OVERLAY',
  CLOSE_ALL_OVERLAY = '@@overlay/Container/CLOSE_ALL_OVERLAY',
  UPDATE_PROPS = '@@overlay/Container/UPDATE_PROPS',
}

export interface OverlayContainerState {
  readonly currentOverlays: Overlay[];
}

export type OverlayContainerActionType = ReturnType<
  | typeof openOverlayAction
  | typeof closeOverlayAction
  | typeof closeAllOverlayAction
  | typeof updatePropsAction
>;
