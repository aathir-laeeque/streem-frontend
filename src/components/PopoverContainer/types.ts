import {
  openPopoverAction,
  closePopoverAction,
  closeAllPopoverAction,
} from './actions';

export enum PopoverNames {
  TASK_USER_ASSIGNMENT = 'TASK_USER_ASSIGNMENT',
}

export interface Popover {
  type: string;
  popOverAnchorEl: Element | ((element: Element) => Element);
  props?: Record<string, any>;
}

export interface PopoverContainerProps {
  currentPopovers: Popover[];
  closePopover: () => void;
  closeAllPopovers: () => void;
}

export enum PopoverContainerAction {
  OPEN_POPOVER = '@@popover/Container/OPEN_POPOVER',
  CLOSE_POPOVER = '@@popover/Container/CLOSE_POPOVER',
  CLOSE_ALL_POPOVER = '@@popover/Container/CLOSE_ALL_POPOVER',
}

export interface PopoverContainerState {
  readonly currentPopovers: Popover[];
}

export type PopoverContainerActionType = ReturnType<
  | typeof openPopoverAction
  | typeof closePopoverAction
  | typeof closeAllPopoverAction
>;
