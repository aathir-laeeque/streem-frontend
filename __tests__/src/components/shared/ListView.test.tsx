import { InfiniteListView } from '#components';
import { jobs, jobsProperties } from '#mocks';
import { Job } from '#views/Jobs/types';
import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

describe('<InfiniteListView />', () => {
  let snapshot: ReactTestRenderer;
  const data: Job[] = jobs.data;
  beforeAll(() => {
    const component = (
      <InfiniteListView
        properties={jobsProperties.data}
        fetchData={() => jest.fn()}
        isLast={true}
        currentPage={0}
        data={data}
        primaryButtonText="Primary Button"
      />
    );
    snapshot = create(component);
  });

  it('should render and match the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
