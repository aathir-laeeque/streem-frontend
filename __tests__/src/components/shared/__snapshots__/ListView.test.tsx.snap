// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListViewComponent /> should render and match the snapshot 1`] = `
<div
  className="sc-fzoLag ikcvLg"
>
  <div
    className="list-options"
  >
    <button
      className="sc-fzqBZW cHkWYl"
    >
      Filters 
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "color": "#12aab3",
            "fontSize": 20,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </button>
    <div
      className="searchboxwrapper"
    >
      <input
        className="searchbox"
        placeholder="Search"
        type="text"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root searchsubmit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <span
      className="resetOption"
    >
      Reset
    </span>
    <button
      className="sc-fzokOt gDhfqG"
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": 0,
        }
      }
    >
      Primary Button
    </button>
  </div>
  <div
    className="list-header"
  >
    <div
      className="list-header-columns"
    >
      PRODUCT MANUFACTURED
    </div>
    <div
      className="list-header-columns"
    >
      BATCH NO
    </div>
    <div
      className="list-header-columns"
    >
      ROOM ID
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-card"
    >
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
    </div>
    <div
      className="list-card"
    >
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
    </div>
    <div
      className="list-card"
    >
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
    </div>
    <div
      className="list-card"
    >
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
      <div
        className="list-card-columns"
      >
        -N/A-
      </div>
    </div>
  </div>
</div>
`;
